### ISSUE TITLE
Rolls are blocked due to timeouts on Flutter "Linux Web Framework tests"

### ISSUE THREAD

**Issue by alexmarkov** [dart-lang/sdk#51056]
Dart SDK -> engine rolls are blocked because "Linux Web Framework tests" shard fails with a lot of timeouts.

Example log:
```
00:00 +0 -1: compiling /b/s/w/ir/x/w/flutter/packages/flutter/test/rendering/sliver_persistent_header_test.dart [E]                                                                                    
  PathNotFoundException: Cannot open file, path = '/b/s/w/ir/x/w/recipe_cleanup/builder/src/out/host_debug_unopt/dart-sdk/lib/dev_compiler/kernel/amd/require.js' (OS Error: No such file or directory, errno = 2)
  dart:io/common.dart 55:9      _checkForErrorResponse
  dart:io/file_impl.dart 364:7  _File.open.<fn>
  dart:async/zone.dart 1406:47  _rootRunUnary
  ===== asynchronous gap ===========================
  dart:io/file_impl.dart 136:5  _FileStream._start.onOpenFile
  

To run this test again: /b/s/w/ir/x/w/flutter/bin/cache/dart-sdk/bin/dart test /b/s/w/ir/x/w/flutter/packages/flutter/test/rendering/sliver_persistent_header_test.dart -p chrome --plain-name 'compiling /b/s/w/ir/x/w/flutter/packages/flutter/test/rendering/sliver_persistent_header_test.dart'

12:00 +0 -2: compiling /b/s/w/ir/x/w/flutter/packages/flutter/test/rendering/positioned_box_test.dart [E]                                                                                              
  TimeoutException after 0:12:00.000000: Test timed out after 12 minutes.
  package:test_api/src/backend/invoker.dart 334:28  Invoker._handleError.<fn>
  dart:async/zone.dart 1390:47                      _rootRun
  dart:async/zone.dart 1300:19                      _CustomZone.run
  package:test_api/src/backend/invoker.dart 332:10  Invoker._handleError
  package:test_api/src/backend/invoker.dart 288:9   Invoker.heartbeat.<fn>.<fn>
  dart:async/zone.dart 1398:13                      _rootRun
  dart:async/zone.dart 1300:19                      _CustomZone.run
  package:test_api/src/backend/invoker.dart 287:38  Invoker.heartbeat.<fn>
  dart:async-patch/timer_patch.dart 18:15           Timer._createTimer.<fn>
  dart:isolate-patch/timer_impl.dart 398:19         _Timer._runTimers
  dart:isolate-patch/timer_impl.dart 429:5          _Timer._handleMessage
  dart:isolate-patch/isolate_patch.dart 192:26      _RawReceivePort._handleMessage

```
https://logs.chromium.org/logs/flutter/buildbucket/cr-buildbucket/8791626204200190129/+/u/run_test.dart_for_web_tests_shard_and_subshard_0/test_stdout
https://ci.chromium.org/p/flutter/builders/try/Linux%20Web%20Framework%20tests/22619

This started happening between https://github.com/dart-lang/sdk/commit/831db9410537 and https://github.com/dart-lang/sdk/commit/21cb5c1e7273

List of suspected commits:
https://dart.googlesource.com/sdk.git/+log/0097dcbada06..831db9410537

@sigmundch @nshahan @srujzs

**Comment by nshahan** [comment]
My change caused this error. I thought I already prepared for it in the flutter repo with this change https://github.com/flutter/flutter/pull/118120.

I guess there is another location where the path to that file is hard coded. In this case the fix is very simple. We just need to update the path from `dart-sdk/lib/dev_compiler/kernel/amd/require.js` to `dart-sdk/lib/dev_compiler/amd/require.js`. I'll start looking at that test to see where it gets the path from.

If we roll back the change in the SDK we will need to handle the other packages that I have already updated to handle the new location. Golem will be particularly difficult because the build and tar operations there are especially brittle. the build_web_compilers package has already published a version to handle the new location as well.
